### PR TITLE
Prep for pulpcore/3.25 compatibility.

### LINKED ISSUE #770: Ensure compatibility with pulpcore 3.25
Raise lowerbounds requirements of pulpcore and have the necessary django4 migration.

### PR DESCRIPTION
**Pull request by hstct**
closes #770

### CODE REVIEW

**Comment by quba42** [comment]
@mdellweg @dralley @ggainey Any chance any of you could have a look at this? (Since you were involved in the equivalent pulp_rpm change https://github.com/pulp/pulp_rpm/pull/3147).

File: CHANGES/770.feature [hunk 1]
```diff
@@ -0,0 +1 @@
+Declare and require at least pulpcore/3.25 compatibility.
```
Comment (quba42) [inline CHANGES/770.feature:1]: Should this be `CHANGES/770.feature` or perhaps something like `CHANGES/770.misc`?

Maybe it's not bad to have it prominently in the release notes, it is after all the main reason for having the next release. => Argument for feature.

Comment (hstct) [reply CHANGES/770.feature:1]: I'm leaning towards feature for the visibility alone. Can't hurt to announce it at the most prevalent place in addition to release announcements.

Comment (mdellweg) [reply CHANGES/770.feature:1]: Yes to feature

**Comment by mdellweg** [comment]
You want to set this in the plugin template:
run_pulpcore_tests_for_plugins: false

**Review by mdellweg — APPROVED** [review]
Looking good!

**Review by quba42 — APPROVED** [review]
LGTM, though I should add that I have no real understanding of the background why the new Django version needs this new migration even though nothing changed. :wink:

**Comment by ggainey** [comment]
> LGTM, though I should add that I have no real understanding of the background why the new Django version needs this new migration even though nothing changed. wink

iirc, there used-to-be a related_name field that isn't, anymore, and there was a case-change. Both of these are "harmless", but they look different to django's makemigrations --check

**Pull request merged**